### DISCUSSION TITLE
Session CPU usage

### DISCUSSION THREAD

**Discussion by safwanadnan** [devlikeapro/waha#707] [Q&A]
The website states that a 4 GB server can handle around 50 NOWEB sessions. I am running my Waha instance on a 6 GB vps which I estimate can roughly handle around 75 sessions. Now, I wanted to ask if I exceed the number of sessions recommended, what are the potential problems this can cause? Also, most of my sessions are in the stopped state and logged out, so does that also take up any CPU usage or should I just take into calculation the sessions that are actively running? This is very important for me to manage the scalability of my application.

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
>  if I exceed the number of sessions recommended, what are the potential problems this can cause

The sesssions will be in "reconnection" cycle over and over again (because it doesn't have CPU time to send "ping" messages), you'll like loose messages or authentication (so you'll have to rescan QR for session to re-pair session) and the API will work slow

> most of my sessions are in the stopped state and logged out

STOPPED one doesn't consume CPU, so that's fine! 

>  take into calculation the sessions that are actively running?

Correct, only sessions in **"running"** block consume resources (CPU and Memory) :point_down: 
https://waha.devlike.pro/docs/how-to/sessions/
![image](https://github.com/user-attachments/assets/b10ebcee-c2ed-4937-a036-b48de3382667)


[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
btw, we recommend doing "sharding" horizontal scaling like https://waha.devlike.pro/blog/waha-scaling-how-to-handle-500-sessions/#horizontal-scaling---sharding

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
